### PR TITLE
feat: Pagerduty incident notes

### PR DESCRIPTION
**Pull request by shimonp21**

### CODE REVIEW

File: website/pages/blog/announcing-pagerduty-plugin.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
```
Comment (hermanschaaf) [inline website/pages/blog/announcing-pagerduty-plugin.md:1]: Maybe let's separate the blog post into a separate PR, if that's okay?

Comment (shimonp21) [reply website/pages/blog/announcing-pagerduty-plugin.md]: Yeah obviously, I `checkout -b`ed from the wrong branch :)

Comment (shimonp21) [reply website/pages/blog/announcing-pagerduty-plugin.md]: adventures in
```bash
git rebase
git reset --mixed HEAD~
git reset --hard main
```

File: plugins/source/pagerduty/codegen/recipes/base.go [hunk 2]
```diff
@@ -45,6 +45,7 @@ type Resource struct {
 	ResponseStructOverride        string
 	ResponseFieldOverride         string
 	RestPathOverride              string
+	ParentIsPointer               bool
```
Comment (hermanschaaf) [inline plugins/source/pagerduty/codegen/recipes/base.go:48]: I'm not a big fan of this... but since changing to something that doesn't require a field like this will be quite a lot of work, I'm willing to look the other way 😄

**Review by hermanschaaf — APPROVED** [review]
